### PR TITLE
For newer pymongo

### PR DESCRIPTION
**Pull request by richafrank**
Connection.disconnect -> MongoClient.close
Database.connection -> Database.client

See https://api.mongodb.com/python/current/migrate-to-pymongo3.html#the-disconnect-method-is-removed

### CODE REVIEW

**Review by jdricklefs — APPROVED** [review]
Wow, thanks for being thorough. 

Do we need to do something to the version number in setup.py? We broke our internal wheels repo last time we updated this.

**Comment by richafrank** [comment]
> Do we need to do something to the version number in setup.py? We broke our internal wheels repo last time we updated this.

Well we haven't pushed our recent changes upstream for publishing to PyPI under the maintainer's versioning. That would be good, but I didn't want to block on it, since upstream has different pymongo 3 upgrades... 

I think we're installing from a commit here and probably shouldn't build wheels from non-release tags for libs generally?

**Pull request merged**
